acceptance/tests/connect/connect_helper.go

```diff
@@ -17,40 +17,85 @@ import (
-// mesh connectivity.
-func ConnectInjectConnectivityCheck(t *testing.T, ctx environment.TestContext, cfg *config.TestConfig, secure bool, autoEncrypt bool, cli bool) {
-	helmValues := map[string]string{
-		"connectInject.enabled": "true",
+// ConnectHelper configures a Consul cluster for connect injection tests.
+// It also provides helper methods to exercise the connect functionality.
+type ConnectHelper struct {
+	// ClusterKind is the kind of Consul cluster to use (e.g. "Helm", "CLI").
+	ClusterKind consul.ClusterKind
 
-		"global.tls.enabled":           strconv.FormatBool(secure),
-		"global.tls.enableAutoEncrypt": strconv.FormatBool(autoEncrypt),
-		"global.acls.manageSystemACLs": strconv.FormatBool(secure),
-	}
+	// Secure configures the Helm chart for the test to use ACL tokens.
+	Secure bool
 
-	releaseName := helpers.RandomName()
-	var consulCluster consul.Cluster
-	if cli {
-		consulCluster = consul.NewCLICluster(t, helmValues, ctx, cfg, consul.CLIReleaseName)
-	} else {
-		consulCluster = consul.NewHelmCluster(t, helmValues, ctx, cfg, releaseName)
+	// AutoEncrypt configures the Helm chart for the test to use AutoEncrypt.
+	AutoEncrypt bool
+
+	// HelmValues are the additional helm values to use when installing or
+	// upgrading the cluster beyond connectInject.enabled, global.tls.enabled,
+	// global.tls.enableAutoEncrypt, global.acls.mangageSystemACLs which are
+	// set by the Secure and AutoEncrypt fields.
+	HelmValues map[string]string
+
+	// RelaseName is the name of the Consul cluster.
+	ReleaseName string
+
+	T   *testing.T
+	Ctx environment.TestContext
+	Cfg *config.TestConfig
+
+	// consulCluster is the cluster to use for the test.
+	consulCluster consul.Cluster
+
+	// consulClient is the client used to test service mesh connectivity.
+	consulClient *api.Client
+}
+
+// Setup creates a new cluster using the New*Cluster function and assigns it
+// to the consulCluster field.
+func (c *ConnectHelper) Setup() {
+	switch c.ClusterKind {
+	case consul.Helm:
+		c.consulCluster = consul.NewHelmCluster(c.T, c.helmValues(), c.Ctx, c.Cfg, c.ReleaseName)
+	case consul.CLI:
+		c.consulCluster = consul.NewCLICluster(c.T, c.helmValues(), c.Ctx, c.Cfg, c.ReleaseName)
 	}
+}
 
-	consulCluster.Create(t)
+// Install uses the consulCluster to install Consul onto the Kubernetes cluster.
+func (c *ConnectHelper) Install() {
+	logger.Log(c.T, "Installing Consul cluster")
+	c.consulCluster.Create(c.T)
+	c.consulClient = c.consulCluster.SetupConsulClient(c.T, c.Secure)
+}
+
+// Upgrade uses the existing Consul cluster and upgrades it using Helm values
+// set by the Secure, AutoEncrypt, and HelmValues fields.
+func (c *ConnectHelper) Upgrade() {
+	require.True(c.T, c.consulCluster != nil, "consulCluster must be set before calling Upgrade (Call Install first).")
+	require.True(c.T, c.consulClient != nil, "consulClient must be set before calling Upgrade (Call Install first).")
```

Comment:
you can use `require.NotNil` here instead of true. It will make the assertion more readable.